internal/config/config_test.go

```diff
@@ -177,3 +181,57 @@ func TestConfig_LoadDirectory(t *testing.T) {
 	assert.Equal(t, pConfig, c.Inputs[3].Config,
 		"Merged Testdata did not produce correct procstat metadata.")
 }
+
+func TestConfig_LoadSpecialTypes(t *testing.T) {
+	c := NewConfig()
+	err := c.LoadConfig("./testdata/special_types.toml")
+	assert.NoError(t, err)
+	require.Equal(t, 1, len(c.Inputs))
+
+	inputHTTPListener, ok := c.Inputs[0].Input.(*http_listener_v2.HTTPListenerV2)
+	assert.Equal(t, true, ok)
+	assert.Equal(t, internal.Duration{Duration: time.Second}, inputHTTPListener.WriteTimeout)
+	assert.Equal(t, internal.Size{Size: 1024 * 1024}, inputHTTPListener.MaxBodySize)
+	assert.Equal(t, "/path/to/my/cert\n", inputHTTPListener.TLSCert)
+
+	c = NewConfig()
+	err = c.LoadConfig("./testdata/invalid_field.toml")
+	require.Error(t, err, "invalid field name")
+	// commented out asserts are differences in error output from influxdata/toml and naoina/toml
```

Comment:
These all look to be pretty reasonable changes to the error string, go ahead and remove these comments.